pkg/webhook/cspc.go

```diff
@@ -128,6 +131,37 @@ func (wh *webhook) validateCSPCCreateRequest(req *v1beta1.AdmissionRequest) *v1b
 	return response
 }
 
+// validateCSPCDeleteRequest validates CSPC delete request
+// if any cvrs exist on the cspc pools then deletion is invalid
+func (wh *webhook) validateCSPCDeleteRequest(req *v1beta1.AdmissionRequest) *v1beta1.AdmissionResponse {
+	response := NewAdmissionResponse().SetAllowed().WithResultAsSuccess(http.StatusAccepted).AR
+	cspiList, err := cspi.NewKubeClient().WithNamespace(req.Namespace).List(
+		metav1.ListOptions{
+			LabelSelector: string(apis.CStorPoolClusterCPK) + "=" + req.Name,
+		})
+	if err != nil {
+		klog.Errorf("Could not list cspi for cspc %s: %s", req.Name, err.Error())
+		response = BuildForAPIObject(response).UnSetAllowed().WithResultAsFailure(err, http.StatusBadRequest).AR
+		return response
+	}
+	for _, cspiObj := range cspiList.Items {
+		cvrList, err := cvr.NewKubeclient().WithNamespace(cspiObj.Namespace).List(metav1.ListOptions{
```

Comment:
CVRs seems to be created in openebs namespace rather than cspiObj namespace.. so, list can fail here

Reply:
As the same container is having controllers for `CSPI` and `CVR`, shouldn't they be created in same namespace?